### PR TITLE
replace onnx:: with ONNX_NAMESPACE::

### PR DESCRIPTION
**Pull request by yangchen-MS**
replace hard-coded namespace qualifier onnx:: with the macro ONNX_NAMESPACE::

### CODE REVIEW

**Comment by snnn** [comment]
Why it is needed?

File: cmake/onnxruntime_unittests.cmake [hunk 1]
```diff
@@ -623,6 +623,7 @@ if (onnxruntime_BUILD_SHARED_LIB)
           protobuf::libprotobuf
           DEPENDS ${all_dependencies}
   )
+  target_compile_definitions(onnxruntime_shared_lib_test PUBLIC "ONNX_NAMESPACE=onnx")
```
Comment (snnn) [inline cmake/onnxruntime_unittests.cmake]: Is it needed by anything?

File: onnxruntime/test/shared_lib/test_tensor_loader.cc [hunk 2]
```diff
@@ -17,11 +17,11 @@ namespace test {
 
 TEST_F(CApiTest, load_simple_float_tensor_not_enough_space) {
   // construct a tensor proto
-  onnx::TensorProto p;
+  ONNX_NAMESPACE::TensorProto p;
```
Comment (snnn) [inline onnxruntime/test/shared_lib/test_tensor_loader.cc]: This file doesn't need be changed

File: onnxruntime/test/shared_lib/test_model_loading.cc [hunk 3]
```diff
@@ -14,7 +14,7 @@ namespace {
 void WriteStringToTempFile(const char* test_data, std::basic_string<ORTCHAR_T>& filename) {
```
Comment (snnn) [inline onnxruntime/test/shared_lib/test_model_loading.cc]: This file doesn't need be changed

**Comment by yangchen-MS** [comment]
> 
> 
> Why it is needed?

We have an internal build that needs to be in a namespace different from onnx::

File: cmake/onnxruntime_unittests.cmake (hunk 1 above)
Comment (yangchen-MS) [reply cmake/onnxruntime_unittests.cmake]: It's needed by test_tensor_loader.cc. If we don't want to change test_tensor_loader.cc, then we don't need this change.

File: onnxruntime/test/shared_lib/test_tensor_loader.cc (hunk 2 above)
Comment (yangchen-MS) [reply onnxruntime/test/shared_lib/test_tensor_loader.cc]: Out of curiosity, why we don't want to make the changes for these tests, test_tensor_loader.cc and test_model_loading.cc?

Comment (snnn) [reply onnxruntime/test/shared_lib/test_tensor_loader.cc]: Even you don't change them, it shouldn't matter. It won't break your internal build.

Comment (yangchen-MS) [reply onnxruntime/test/shared_lib/test_tensor_loader.cc]: OK, that's a fair point. I will revert the changes to these two tests and the cmake file.

Comment (yangchen-MS) [reply onnxruntime/test/shared_lib/test_tensor_loader.cc]: Actually, I tend to keep the change to test_tensor_loader.cc to make it consistent with the rest of the file, where all but this one use ONNX_NAMESPACE. What do you think?

Comment (snnn) [reply onnxruntime/test/shared_lib/test_tensor_loader.cc]: Which is the 'rest'?

Comment (snnn) [reply onnxruntime/test/shared_lib/test_tensor_loader.cc]: You mean, some file in this dir is using the ONNX_NAMESPACE macro?

Comment (yangchen-MS) [reply onnxruntime/test/shared_lib/test_tensor_loader.cc]: My bad. I was looking at a previous version of my OWN file. I've reverted the change.

**Comment by yangchen-MS** [comment]
@snnn Changming, I updated the PR. Do you have more feedback? Thanks!

**Pull request merged**